MagaluCloud/magalu
# feat(login): login in a headless environment

## Description
We want to be able to login in a headless environment (without a browser installed). To do this, we print out the URL, copy it to to another computer and paste the response URL into the client.

## Related task
- closes #759

How to test
- Create a Dockerfile on the root of your cli repository with this content

```dockerfile
FROM golang:1.21.6-alpine3.19

WORKDIR /app

COPY . .

WORKDIR /app/mgc/cli
RUN go install
RUN go build -o mgc .
```

- Build the image with `docker build -t testecli .`
- Create a container and run sh on it with `docker run --rm -it testecli sh`
- Run `go run . auth login`. The login URL will be printed in the terminal, so you can copy it to a browser
![Captura de tela de 2024-02-06 15-04-34](https://github.com/MagaluCloud/magalu/assets/110136151/cffa6d87-dd84-4b31-b73d-3996c7481c8c)
- Login normally. You'll see a error page, but it's expected. Just copy the new URL and paste it on the terminal
![Captura de tela de 2024-02-06 15-06-17](https://github.com/MagaluCloud/magalu/assets/110136151/96f55e82-7cbd-4e98-8da9-49663eb93b11)

You are supossed to see a output like this
![Captura de tela de 2024-02-05 19-21-01](https://github.com/MagaluCloud/magalu/assets/110136151/7c12285e-7aa5-439f-a082-b280227f5626)
